setup.py

```diff
@@ -110,6 +114,12 @@ def make_cuda_ext(name, module, sources):
         })
 
 
+def make_cpp_ext(name, module, sources):
```

Comment:
Could we merge `make_cpp_ext` and `make_cuda_ext` into one function? This would simplify `__main__`.

Reply:
did that, please see my new commit

Reply:
I suggest determining which extension to use by checking if `.cu` is in the sources? Then there is no need to pass `is_cuda_ext` into function.